### ISSUE TITLE
helm v3 test failure returns exit code 0

### ISSUE THREAD

**Issue by gnalsa** [helm/helm#5990]
I believe there is a regression as this was previously fixed: https://github.com/helm/helm/pull/2425

```
$ helm version
version.BuildInfo{Version:"v3.0.0-alpha.1", GitCommit:"b9a54967f838723fe241172a6b94d18caf8bcdca", GitTreeState:"clean"}
$ helm test run test
RUNNING: test-nginx-test-nginx-5rvxc
FAILED: test-nginx-test-nginx-5rvxc, run `kubectl logs test-nginx-test-nginx-5rvxc --namespace test` for more info
$ echo $?
0
```

**Comment by bacongobbler** [comment]
#2425 was merged in for Helm v2.4.2 - well before Helm 3 was even discussed so I don't think that patch will fix the issue. That being said, we should definitely look into this.

**Comment by bacongobbler** [comment]
Seems to be fixed.

```
><> ./bin/helm install vault incubator/vault
NAME: vault
LAST DEPLOYED: 2019-08-13 16:36:48.55607632 -0700 PDT m=+0.657705011
NAMESPACE: default
STATUS: deployed

NOTES:
1. Get the application URL by running these commands:
  export POD_NAME=$(kubectl get pods --namespace default -l "app=vault" -o jsonpath="{.items[0].metadata.name}")
  echo "Use http://127.0.0.1:8200 as the Vault address after forwarding."
  kubectl port-forward --namespace default  $POD_NAME 8200:8200
><> ./bin/helm test run vault
Pod vault-vault-status-test pending
Pod vault-vault-status-test pending
Pod vault-vault-status-test running
Error: pod vault-vault-status-test failed
><> echo $?
1
><> ./bin/helm version
version.BuildInfo{Version:"v3.0+unreleased", GitCommit:"d99edbfb3ea07113ec1d1aa21487853ad35e0222", GitTreeState:"clean", GoVersion:"go1.12.7"}
```
